src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.db.Storage.people;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public static final int NEGATIVE_AGE = -10;
+    public static final int ZERO_AGE = 0;
+    public static final int BELOW_MINIMUM_AGE = 17;
+    public static final int MINIMUM_AGE = 18;
+    public static final int OVER_MINIMUM_AGE = 21;
+    public static final int EXPECTED_TWO = 2;
+    public static final int EXPECTED_FOUR = 4;
+    public static final int FIRST = 0;
+    public static final String LOGIN_YURI = "mylogin";
+    public static final String LOGIN_DASHA = "DZlogin";
+    public static final String LOGIN_NAZAR = "nazar2005";
+    public static final String CORRECT_LOGIN = "qwerty";
+    public static final String INCORRECT_LOGIN = "log";
+    public static final String CORRECT_LOGIN_SIX_LETTERS = "mlogin";
+    public static final String CORRECT_PASSWORD = "qwertyu";
+    public static final String CORRECT_PASSWORD_SIX_LETTERS = "123456";
+    public static final String INCORRECT_PASSWORD = "qwer";
+    public static final String EMPTY_LINE = "";
+
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        people.clear();
+    }
+
+    @Test
+    void register_nullLoginValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(null, CORRECT_PASSWORD, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_NullPasswordValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, null, MINIMUM_AGE)));
+    }
+
+    @Test
+    void register_NullAgeValue_NotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(
+                new User(LOGIN_YURI, CORRECT_PASSWORD, null)));
+    }
+
+    @Test
+    void register_correctEge_Ok() {
```

Comment:
```suggestion
    void register_correctAge_Ok() {
```


Methods correctAge, correctPassword, correctLogin are redundant. You should test all possible notOk cases and validUser case with all valid parameters